src/math/math_base.h

```diff
@@ -105,5 +106,73 @@ struct PointBase3D : public PointBase2D<_Type>
     _Type z;
 };
 
+template <typename _Type>
+struct LineBase2D
+{
+    LineBase2D(const PointBase2D<_Type> & _p1, const PointBase2D<_Type> & _p2)
+        : p1( _p1 ), p2( _p2 )
+    { }
+
+    LineBase2D(const PointBase2D<_Type> & _p, double angleRadians)
+    {
+        this->p1 = PointBase2D<_Type>(_p.x, _p.y);
+        this->p2 = PointBase2D<_Type>(std::cos(angleRadians) * length(), std::sin(angleRadians) * length());
+    }
+
+    bool operator == ( const LineBase2D & line ) const
+    {
+        return p1 == line.p1 && p2 == line.p2;
```

Comment:
This function is invalid as two lines might have different points but all 4 points in result sit on one line.

Reply:
Even with four points having the same orientation, if we represent two lines traversing each pair of points, they are still geometrically equal. Not sure how you want me to implement this but feel free to change it.

Reply:
If we have implemented `distance` function then comparing angles of both lines and measuring the distance from one point of line to another line will give us the correct result.